rllib/core/rl_trainer/trainer_runner.py

```diff
@@ -100,19 +106,103 @@ def __init__(
     def is_local(self) -> bool:
         return self._is_local
 
-    def update(self, batch: MultiAgentBatch) -> List[Mapping[str, Any]]:
-        """Do a gradient based update to the RLTrainer(s) maintained by this TrainerRunner.
+    def fit(
+        self,
+        batch: MultiAgentBatch,
+        *,
+        minibatch_size: int,
+        num_iters: int,
+        reduce_fn: Optional[Callable[[ResultDict], ResultDict]] = _reduce_mean_results,
+    ) -> Mapping[str, Any]:
+        """Do `num_iters` minibatch updates given the original batch.
+
+        Given a batch of episodes you can use this method to take more
+        than one backward pass on the batch. The same minibatch_size and num_iters gets
+        will be used for all module ids (previously known as policies) in the
+        multiagent batch
 
         Args:
             batch: The data to use for the update.
+            minibatch_size: The size of the minibatch to use for each update.
+            num_iters: The number of complete passes over all the sub-batches
+                in the input multi-agent batch.
+            reduce_fn: See `update()` documenation for more details.
+
+        Returns:
+            A dictionary of results summarizing the statistics of the updates.
+        """
+
+        start = {mid: 0 for mid in batch.policy_batches.keys()}
+        num_covered_epochs = {mid: 0 for mid in batch.policy_batches.keys()}
+        results = []
+        # TODO (Kourosh): One data transfer is probably better than many for each mini
+        # batch. How should we do this?
+        # loop until the number of passes through all modules batches reaches the
+        # num_iters
+        while min(num_covered_epochs.values()) < num_iters:
+            minibatch = {}
+            for module_id, module_batch in batch.policy_batches.items():
+                s = start[module_id]  # start
```

Comment:
dude, module_id is a number? why reuse these as indices?

Reply:
no?! it's a str? start is a mapping dict from module_id to the start index of the batch of each policy. I don't understand your comment.

Reply:
oh i see, never mind.